src/test/java/com/opentable/versionedconfig/GitServiceTest.java

```diff
@@ -172,15 +176,30 @@ public void testRevisionMetadata() throws IOException {
     public void testCredentialsMustBothBeEmptyOrNull() {
         assertThatExceptionOfType(IllegalArgumentException.class)
                 .isThrownBy(() -> VersioningService.forGitRepository(
-                        new GitProperties(null, "username", null, null, "master")))
+                        new GitProperties(Collections.emptyList(), "username", null, null, "master")))
                 .withMessageContaining("must provide username and password");
 
         assertThatExceptionOfType(IllegalArgumentException.class)
                 .isThrownBy(() -> VersioningService.forGitRepository(
-                        new GitProperties(null, null, "password", null, "master")))
+                        new GitProperties(Collections.emptyList(), null, "password", null, "master")))
                 .withMessageContaining("must provide username and password");
     }
 
+    @Test
+    public void testFallbackPull() throws Exception {
+        workFolder.create();
+        final File checkoutSpot = workFolder.newFolder("init");
+        final GitProperties gitProperties = new GitProperties(ImmutableList.of(URI.create("git://example.invalid"), remote.getLocalPath().toUri()), null, null, checkoutSpot, "master");
+        try (final VersioningService service = new GitService(gitProperties)) {
+            final Optional<VersionedConfigUpdate> maybeVcu = service.checkForUpdate();
+            assertThat(maybeVcu).isPresent();
+            final VersionedConfigUpdate vcu = maybeVcu.get();
+            assertThat(vcu.getOldRevisionMetadata()).isEqualTo(ObjectId.zeroId());
+            assertThat(vcu.getNewRevisionMetadata()).isNotNull();
+            assertThat(changeNames(vcu)).contains("foo.txt");
```

Comment:
I feel like this set of assertions would benefit from being wrapped up as a single "this is the update we expect" assertion, just to provide more hints to readers about what they collectively mean. Took me a while to unpick the meaning of this, although not used to reading Java..

Reply:
I concur, a working feature covered by a valid test == shipit, other improvements can come later if at all.